app_dart/test/foundation/utils_test.dart

```diff
@@ -180,5 +181,56 @@ void main() {
         expect(tableDataList.totalRows, '1');
       });
     });
+
+    group('getFilteredBuilders', () {
+      List<String> files;
+      List<Map<String, dynamic>> builders;
+
+      test('does not return builders when run_if does not match any file', () async {
+        builders = (json.decode('[{"name": "abc", "repo": "def", "task_name": "ghi", "flaky": true, "run_if": ["d/"]}]')
+                as List<dynamic>)
+            .cast<Map<String, dynamic>>();
+        files = <String>['a/b', 'c/d'];
+        final List<Map<String, dynamic>> result = await getFilteredBuilders(builders, files);
+        expect(result.length, 0);
+      });
+
+      test('returns builders when run_if is null', () async {
+        files = <String>['a/b', 'c/d'];
+        builders = (json.decode('[{"name": "abc", "repo": "def", "task_name": "ghi", "flaky": true}]') as List<dynamic>)
+            .cast<Map<String, dynamic>>();
+        final List<Map<String, dynamic>> result = await getFilteredBuilders(builders, files);
+        expect(result, builders);
+      });
+
+      test('returns builders when run_if matches files', () async {
+        files = <String>['a/b', 'c/d'];
+        builders = (json.decode('[{"name": "abc", "repo": "def", "task_name": "ghi", "flaky": true, "run_if": ["a/"]}]')
+                as List<dynamic>)
+            .cast<Map<String, dynamic>>();
+        final List<Map<String, dynamic>> result = await getFilteredBuilders(builders, files);
+        expect(result, builders);
+      });
+
+      test('returns builders when run_if matches files with **', () async {
+        builders =
+            (json.decode('[{"name": "abc", "repo": "def", "task_name": "ghi", "flaky": true, "run_if": ["a/**"]}]')
+                    as List<dynamic>)
+                .cast<Map<String, dynamic>>();
+        files = <String>['a/b', 'c/d'];
+        final List<Map<String, dynamic>> result = await getFilteredBuilders(builders, files);
+        expect(result, builders);
+      });
+
+      test('returns builders when run_if matches files with both * and **', () async {
+        builders =
```

Comment:
We need one more test for file and folder with the same name "a" and a/b/c. E.g a run_if = ["a"] does not match a files changes "a/b/c"

Reply:
Added.